### PR TITLE
add KVTableFactory

### PR DESCRIPTION
**Pull request by bxq2011hust**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- test/unittests/libprecompiled/test_SimpleTablePrecompiled.cpp  8
- libprecompiled/TablePrecompiled.h  1
- libprecompiled/SimpleTablePrecompiled.cpp  2
- libprecompiled/TableFactoryPrecompiled.h  1
- libprecompiled/TablePrecompiled.cpp  2
- test/unittests/libprecompiled/test_SimpleTableFactoryPrecompiled.cpp  3
- libprecompiled/TableFactoryPrecompiled.cpp  3
- libprecompiled/SimpleTableFactoryPrecompiled.cpp  3
- libprecompiled/SimpleTablePrecompiled.h  1
- libprecompiled/Common.cpp  2
- libprecompiled/SimpleTableFactoryPrecompiled.h  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4969127&bid=16416986)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- libprecompiled/SimpleTablePrecompiled.cpp  2
- libprecompiled/SimpleTableFactoryPrecompiled.cpp  3
- test/unittests/libprecompiled/test_SimpleTablePrecompiled.cpp  8
- libprecompiled/SimpleTableFactoryPrecompiled.h  1
- libprecompiled/Common.cpp  2
- libprecompiled/SimpleTablePrecompiled.h  1
- test/unittests/libprecompiled/test_SimpleTableFactoryPrecompiled.cpp  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4969127&bid=16416986)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- libprecompiled/SimpleTableFactoryPrecompiled.cpp  3
- test/unittests/libprecompiled/test_SimpleTableFactoryPrecompiled.cpp  3
- libprecompiled/SimpleTableFactoryPrecompiled.h  1
- libprecompiled/Common.cpp  2
- test/unittests/libprecompiled/test_SimpleTablePrecompiled.cpp  8
- libprecompiled/SimpleTablePrecompiled.h  1
- libprecompiled/SimpleTablePrecompiled.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4969127&bid=16416986)

File: libprecompiled/SimpleTableFactoryPrecompiled.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+std::string SimpleTableFactoryPrecompiled::toString()
+{
+    return "SimpleTableFactory";
+}
+
+bytes SimpleTableFactoryPrecompiled::call(
+    ExecutiveContext::Ptr context, bytesConstRef param, Address const& origin)
+{
+    STORAGE_LOG(TRACE) << LOG_BADGE("SimpleTableFactoryPrecompiled") << LOG_DESC("call")
```
Comment (cyjseagull) [inline libprecompiled/SimpleTableFactoryPrecompiled.cpp]: Recommend to use `DEBUG`

File: libprecompiled/SimpleTablePrecompiled.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+std::string SimpleTablePrecompiled::toString()
+{
+    return "SimpleTable";
+}
+
+bytes SimpleTablePrecompiled::call(
+    ExecutiveContext::Ptr context, bytesConstRef param, Address const& origin)
+{
+    PRECOMPILED_LOG(TRACE) << LOG_BADGE("SimpleTablePrecompiled") << LOG_DESC("call")
```
Comment (cyjseagull) [inline libprecompiled/SimpleTablePrecompiled.cpp]: Recommend to use `DEBUG`

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- libprecompiled/SimpleTableFactoryPrecompiled.cpp  2
- test/unittests/libprecompiled/test_SimpleTableFactoryPrecompiled.cpp  3
- libprecompiled/SimpleTableFactoryPrecompiled.h  1
         

Clones removed
==============
+ libprecompiled/SimpleTablePrecompiled.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4969127&bid=16416986)

File: libprecompiled/SimpleTableFactoryPrecompiled.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+
+using namespace dev;
+using namespace dev::blockverifier;
+using namespace std;
+using namespace dev::storage;
+using namespace dev::precompiled;
+
+const char* const TABLE_METHOD_OPT_STR = "openTable(string)";
+const char* const TABLE_METHOD_CRT_STR_STR = "createTable(string,string,string)";
```
Comment (wheatli) [inline libprecompiled/SimpleTableFactoryPrecompiled.cpp]: TABLE_METHOD_CRT_STR_STR_STR ?

File: libblockverifier/ExecutiveContextFactory.cpp [hunk 4]
```diff
@@ -45,7 +46,9 @@ void ExecutiveContextFactory::initExecutiveContext(
 
     auto tableFactoryPrecompiled = std::make_shared<dev::blockverifier::TableFactoryPrecompiled>();
     tableFactoryPrecompiled->setMemoryTableFactory(memoryTableFactory);
-
+    auto simpleTableFactoryPrecompiled =
```
Comment (wheatli) [inline libblockverifier/ExecutiveContextFactory.cpp]: simpleTable -> KVTable

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- libprecompiled/KVTableFactoryPrecompiled.h  1
- test/unittests/libprecompiled/test_KVTablePrecompiled.cpp  8
- libprecompiled/KVTablePrecompiled.cpp  1
- test/unittests/libprecompiled/test_KVTableFactoryPrecompiled.cpp  3
- libprecompiled/KVTableFactoryPrecompiled.cpp  1
- libprecompiled/KVTablePrecompiled.h  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4969127&bid=16416986)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- test/unittests/libprecompiled/test_KVTableFactoryPrecompiled.cpp  3
- test/unittests/libprecompiled/test_KVTablePrecompiled.cpp  8
- libprecompiled/KVTablePrecompiled.cpp  1
- libprecompiled/KVTablePrecompiled.h  1
- libprecompiled/KVTableFactoryPrecompiled.h  1
- libprecompiled/KVTableFactoryPrecompiled.cpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4969127&bid=16416986)

**Pull request merged**